content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-node-definition-reference.md

```diff
@@ -0,0 +1,382 @@
+---
+linktitle: Script Canvas Node Definition Reference
+title: Script Canvas Node Definition Reference
+description: Reference guide on all the available parameters when defining Script Canvas nodes
+weight: 200
+---
+
+## Introduction
+
+Script Canvas Node Definition files are XML files that AzAutoGen uses to generate the necessary C++ code for registering nodes, defining their topology and reducing the amount of "boilerplate" code that would otherwise be necessary. This reference guide covers all the available XML tags and attributesa needed to create Script Canvas nodes.
+
+---
+
+### ScriptCanvas
+
+The __ScriptCanvas__ tag is the top-level description of the node. It is used to produce the C++ class that represents the node and to configure its serialization & editor properties.
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------|
+| __Include__   | Required |  The name of the Script Canvas Node  | `Include="Include/ScriptCanvas/Internal/Nodeables/BaseTimer.h"` |
+| __xmlns:xsi__ | Recommended | Indicates that the elements and data types used in the schema come from the `"http://www.w3.org/2001/XMLSchema"` namespace. It also specifies that the elements and data types that come from it should be prefixed with xs: | `xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"` |
+
+----
+
+### Class
+
+The __Class__ element is used to define the Node's O3DE configuration, many of these attributes configure O3DE's serialization properties.
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------|------------|
+| __Name__   | Required |  The name of the Script Canvas Node  | `Name="My Node"` |
+| __QualifiedName__  | Required |  Fully qualified name, includes all namespaces, (e.g. ScriptCanvas::Nodes::MyNode) | `QualifiedName="ScriptCanvas::Nodes::MyNode"` |
+| __Description__ | Recommended | Describes the node and its functionality, this is displayed as a tooltip on the node | `Description="An example node"` |
+| __Category__ | Recommended | Category will define where this node will be places on the Node Palette, you can nest categories by separating with the / character (e.g. "My Nodes/Math") | `Category="Math/Linear Algebra"` |
+| __PreferredClassName__ | Optional | In some cases it may be desirable to override the node's class name in the editor, this field does not change functionality, only the node's class name used in the __EditContext__ | `PreferredClassName="A Different Name"`|
+| __Uuid__ | Optional | Allows you to provide a specific Universal Identifier, this is generally generated but in some situations it may be desirable to provide a specific UUID.  | `Uuid="{EE36A690-7C33-445F-B9E8-BD045D6ACC1D}"` |
```

Comment:
Yes, generally this should be empty it's only intended for rare circumstances, for example if a node becomes deprecated it may sometimes be useful to specify a new UUID to remap it